app/client/components/SpeedIndicator.js

```diff
@@ -27,7 +33,7 @@ export default class SpeedIndicator extends React.Component {
         }
 
         return (
-            <div style={ speedStyle }>
+            <div style={ speedStyle } onClick={(e) => {this.speedClicked(e)}}>
```

Comment:
if you copy/paste my code it will work.

## create local symbols

```
var { speedClicked } = this;
```

this code has the purpose to **import** the `speedClicked` method into the `render()` function.
(so you don't have to use `this.` in your code!)

## assign clear callbacks

```
<div style={ speedStyle } onClick={speedClicked}>
```

this piece of code will just assign the imported callback to the `onClick` event.

NOTE: the implementation you used did create an anonymous arrow function that will just call the callback anyway. So it is both memory and cpu that can be saved.